### PR TITLE
Split out batch and single message options

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/SingleMessageOptions.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            {
+                Argument.AssertNotNegative(value, nameof(MaxAutoLockRenewalDuration));
+                _maxAutoRenewDuration = value;
+            }
+        }
+        private TimeSpan _maxAutoRenewDuration = TimeSpan.FromMinutes(5);
+
+        /// <summary>Gets or sets the maximum number of concurrent calls to a function. Note each call
+        /// would be passing a different message. The default is 16 times the return value of
+        /// <see cref="Utility.GetProcessorCount"/>.
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/SingleMessageOptions.cs:36]: `Utility` is internal and not something that customers would have awareness of.   Maybe revise to something like "...  The default is 16 times the count of available processors" ?

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/BatchMessageOptions.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+    /// <summary>
+    /// Configuration options for functions that bind to multiple multiple <see cref="ServiceBusReceivedMessage"/>.
+    /// </summary>
+    public class BatchMessageOptions
+    {
+        /// <summary>
+        /// Gets or sets the maximum number of messages that will be passed to each function call. The default value is 1000.
+        /// </summary>
+        public int MaxMessages { get; set; } = 1000;
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/BatchMessageOptions.cs:17]: Seems odd that there's an exception handler for the single but not the batch...  was that intentional?

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/BatchMessageOptions.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/BatchMessageOptions.cs:17]: Yeah this is just a side effect of the single using the Processor whereas batch uses the Receiver.

**Comment by JoshLove-msft** [comment]
> What did we have in track 1?

BatchOptions, MessageHandlerOptions (from Track 1 SDK), SessionHandlerOptions (from Track 1 SDK)

**Comment by pakrym** [comment]
I'm OK with the change in general but would like to run it past the Functions team during the review. I'm not sure if any of the options we've separated might later be applicable to both.

**Comment by pakrym** [comment]
For example, why doesn't `MaxReceiveWaitTime` apply to the single message?

**Comment by JoshLove-msft** [comment]
Closing for now pending API review of the current surface. Will reopen if we want to go this route.

**Pull request closed without merging**